intersect.go

```diff
@@ -44,6 +44,28 @@ func EveryBy[T any](collection []T, predicate func(item T) bool) bool {
 	return true
 }
 
+// Equivalent Returns true if the subset has the same elements and the same number of each element as the collection.
+func Equivalent[T comparable](collection, subset []T) bool {
+	l := len(collection)
+	if l != len(subset) {
+		return false
+	}
+
+	var m = make(map[T]int, l)
+	for i := range collection {
+		m[collection[i]] += 1
+	}
+	for i := range subset {
+		m[subset[i]] -= 1
+	}
+	for _, v := range m {
+		if v != 0 {
+			return false
+		}
+	}
```

Comment:
Your logic requires extra space in the map and also 3 times iterating over the map.
Why not just sort the 2 arrays and compare if both are the same ?? 

What are your thoughts ??

Also after adding the function, you can add the function to Readme also.
Look this PR as reference : https://github.com/samber/lo/pull/494

Reply:
Because just `!=` or `==` is allowed on generics comparable, we can't sort this slice. 
Additionally, sort.Slice() will change the parameter's original order, which I think is unsafe.